decision-records/0004-data-api.md

```diff
@@ -0,0 +1,113 @@
+# Define how to expose version through the collection metadata API
+
+- Date: 2023-10-25
+
+## Context and Problem Statement
+
+### Context
+
+The need to extend the current API comes as part of our collaboration with Terms of Service; Didn't Read (ToS;DR), whose web application will be adapted to obtain data from public OTA datasets instead of the ToS;DR server database.
+
+Currently, the terms annotated by ToS;DR are currently taken from original web documents stored on the ToS;DR server database. Open Terms Archive provides a significantly higher quality source for terms, for example by cleaning up all content that is not related to the terms, combining documents scattered across several pages, or normalising PDF documents.
+
+To allow the transition from historical ToS;DR's crawler and database to the publicly accessible OTA datasets, ToS;DR requires access to the content of the versions through an API. This RFC outlines the proposed solutions.
+
+## Proposed solutions
+
+### Solution A:
+
+#### Base URL
+
+`<collection host>/api/:version`
+
+#### Endpoint
+
+##### `GET /version/:serviceId/:termsType/:date.md`
+
+Return the markdown content of the version valid for the given date
+
+##### Parameters
+
+| Parameter | Type   | Description            |
+| --------- | ------ | ---------------------- |
+| serviceId | URL-encoded string | The ID of the service |
+| termsType | URL-encoded string | The name of terms type |
+| date | URL-encoded ISO date string | The date for which you want the version. This date should not be in the futur |
+
+Note about the date:
```

Comment:
```suggestion
Note about the date:
- A full date and time is required, and not a simple date (such as `2023-10-24`), to avoid ambiguities on days where a version changed, and timezone differences between client and server.
```